src/main/java/studio/magemonkey/fusion/gui/RecipeGui.java

```diff
@@ -228,132 +228,137 @@ public void initialize() {
     }
 
     public void reloadRecipes() {
-        if (!player.isOnline()) return;
-        try {
-            setPattern();
-
-            /* Default setup */
-            ItemStack          fill       = table.getFillItem();
-            Collection<Recipe> allRecipes = new ArrayList<>(category.getRecipes());
-            allRecipes.removeIf(r -> r.isHidden(player));
-            int pageSize       = resultSlots.size();
-            int allRecipeCount = allRecipes.size();
-            int i              = 0;
-            int page           = this.page;
-
-            int fullPages = allRecipeCount / pageSize;
-            int rest      = allRecipeCount % pageSize;
-            int pages     = (rest == 0) ? fullPages : (fullPages + 1);
-            if (player.isOnline() && page >= pages) {
-                if (page > 0) {
-                    this.page = pages - 1;
-                }
+        Bukkit.getScheduler().runTaskAsynchronously(Fusion.getInstance(), () -> {
```

Comment:
This asynchronous implementation from me before is working, but it happens to make a weird effect as goflish reported. Its loading the items one by one in the GUI, just like a queue. I think this could happen due to the same stuff on the `reloadRecipesTask`. Maybe we should replace `reloadRecipesTask` to `reloadRecipes` and see if it fixes this occurence